src/main.js

```diff
@@ -0,0 +1,192 @@
+    requires: ['profile'],
+    setup: ({profile}) => config({profile}),
+  },
+
+  memcached: {
+    requires: ['cfg'],
+    setup: ({cfg}) => {
+      assert(cfg.memcached.servers, 'Must specify memcached servers');
+      return new Memcached(cfg.memcached.servers);
+    },
+  },
+
+  sqs: {
+    requires: ['cfg'],
+    setup: ({cfg}) => {
+      assert(cfg.sqs, 'Must specify config for SQS');
+      return new aws.SQS(cfg.sqs);
+    },
+  },
+
+  influx: {
+    requires: ['cfg'],
+    setup: ({cfg}) => {
+      if (cfg.influx.connectionString) {
+        return new base.stats.Influx(cfg.influx);
+      }
+      return new base.stats.NullDrain();
+    }
+  },
+
+  monitor: {
+    requires: ['cfg', 'influx', 'process'],
+    setup: ({cfg, influx, process}) => base.stats.startProcessUsageReporting({
+      drain:      influx,
+      component:  cfg.app.statsComponent,
+      process:    process
+    })
+  },
+
+  // Validator and publisher
+  validator: {
+    requires: ['cfg'],
+    setup: ({cfg}) => base.validator({
+      folder:        path.join(__dirname, 'schemas'),
+      constants:     require('./schemas/constants'),
+      publish:       cfg.app.publishMetaData,
+      schemaPrefix:  'cloud-mirror/v1/',
+      aws:           cfg.aws
+    })
+  },
+
+  /*publisher: {
+    requires: ['cfg', 'validator', 'influx', 'process'],
+    setup: ({cfg, validator, influx, process}) => exchanges.setup({
+      credentials:        cfg.pulse,
+      exchangePrefix:     cfg.app.exchangePrefix,
+      validator:          validator,
+      referencePrefix:    'cloud-mirror/v1/exchanges.json',
+      publish:            cfg.app.publishMetaData,
+      aws:                cfg.aws,
+      drain:              influx,
+      component:          cfg.app.statsComponent,
+      process:            process
+    })
+  },*/
+
+  api: {
+    requires: [
+      'cfg', /*'publisher',*/ 'validator', 'influx', 'memcached', 's3backends',
+    ],
+    setup: (ctx) => v1.setup({
+      context: {
+        //publisher: ctx.publisher,
+        validator: ctx.validator,
+        memcached: ctx.memcached,
+        s3backends: ctx.s3backends,
+      },
+      validator: ctx.validator,
+      authBaseUrl: ctx.cfg.taskcluster.authBaseUrl,
+      publish: ctx.cfg.app.publishMetaData,
+      baseUrl: ctx.cfg.server.publicUrl + '/v1',
+      referencePrefix: 'cloud-mirror/v1/api.json',
+      aws: ctx.cfg.aws,
+      component: ctx.cfg.app.statsComponent,
+      drain: ctx.influx
+    })
+  },
+
+  // Create the server process
+  server: {
+    requires: ['cfg', 'api', 'monitor'],
+    setup: ({cfg, api}) => {
+      let app = base.app(cfg.server);
+      app.use('/v1', api);
+      return app.createServer();
+    }
+  },
+
+  s3backends: {
+    requires: ['cfg', 'sqs', 'memcached', 'profile'], 
+    setup: async ({cfg, sqs, memcached, profile}) => {
+      // This should probably not all be here...
+      let s3backends = {};
+      let regions = cfg.backend.s3.regions.split(',');
```

Comment:
You `config.yml` with the default config is missing. (note you can keep your credentials overwriting default in `user-config.yml`).

I note this because you would probably want to use: `env:list S3_REGIONS` inside the config thing... That way it'll load env var `S3_REGIONS` and split it into a list (separated by spaces), see:
https://github.com/taskcluster/typed-env-config